tinygrad/runtime/ops_clang.py

```diff
@@ -1,28 +1,47 @@
-import ctypes, subprocess, pathlib, tempfile
+import os, platform, subprocess, ctypes, pickle
+from mmap import PROT_READ, PROT_WRITE, PROT_EXEC, MAP_ANON, MAP_PRIVATE
 from tinygrad.device import Compiled, Compiler, MallocAllocator
-from tinygrad.helpers import cpu_time_execution, DEBUG, cpu_objdump
+from tinygrad.runtime.support.elf_linker import fixup_relocations
+from tinygrad.helpers import cpu_time_execution, OSX
 from tinygrad.renderer.cstyle import ClangRenderer
 
+libc = ctypes.CDLL(None)
+libc.mmap.restype = ctypes.c_void_p
+libc.mmap.argtypes = (ctypes.c_void_p, ctypes.c_size_t, ctypes.c_int, ctypes.c_int, ctypes.c_int, ctypes.c_size_t)
+
+MAP_JIT = 0x0800 if OSX else 0x0
+
 class ClangCompiler(Compiler):
   def compile(self, src:str) -> bytes:
-    # TODO: remove file write. sadly clang doesn't like the use of /dev/stdout here
-    with tempfile.NamedTemporaryFile(delete=True) as output_file:
-      subprocess.check_output(['clang', '-include', 'tgmath.h', '-shared', '-march=native', '-O2', '-Wall', '-Werror', '-x', 'c', '-fPIC', '-',
-                               '-o', str(output_file.name)], input=src.encode('utf-8'))
-      return pathlib.Path(output_file.name).read_bytes()
+    platspec = ('-ffixed-x18',) if platform.machine() == "arm64" else ('-Xclang=-fnative-half-type', '-Xclang=-fnative-half-arguments-and-returns')
```

Comment:
Are all these flags needed?

Reply:
the way there were added is when i checked on some platform and there were problems. I remember -Xclang=-fnative stuff was required on some relatively old server platform my friend gave me ssh to, -fnative-half-arguments-and-returns was required for something but i'm not totally sure if it still is because now tinymath.h is so tiny (if remove - risk that somebody might run into problems). -ffixed-x18 i explained in monday meeting - it is zeroed on context switches on darwin so we have to tell compiler not to use it because this isn't the case on linux or generic arm